### PR TITLE
chore(ssa refactor): Add all remaining doc comments to ssa generation pass

### PR DESCRIPTION
**Pull request by jfecher**
# Related issue(s)

Resolves # 

# Description

## Summary of changes

Adds a doc comment for every function that was missing one in the ssa generation pass.

There are also some minor code edits:
- Renaming pred_iter and succ_iter to predecessors and successors
- Removing some unused fields
- Using std::mem::take in one spot

## Test additions / changes

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

## Documentation needs
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

File: crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs:71]: Does this read clearer and or cause confusion "If this block has been finalized, this is expected to be Some` ?

File: crates/noirc_evaluator/src/ssa_refactor/ir/cfg.rs
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/ir/cfg.rs:35]:
```suggestion
    /// Compute all of the edges between each block in the function
```

File: crates/noirc_evaluator/src/ssa_refactor/ir/cfg.rs
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/ir/cfg.rs:42]:
```suggestion
    /// Compute all of the edges for the current block given
```

File: crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs:247]: Why is it useful to insert an instruction after a terminator has already been added?

File: crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs:7]: >Note that InstructionIds are not unique.

Do we mean that InstructionIds are unique but they do not refer to unique data? Meaning you can have InstructionId 5 and 6 both referring to the same data

File: crates/noirc_evaluator/src/ssa_refactor/ssa_gen/mod.rs
Comment (kevaundray) [inline crates/noirc_evaluator/src/ssa_refactor/ssa_gen/mod.rs:21]: nit: We use "SSA" and "Ssa"

Comment (kevaundray) [reply crates/noirc_evaluator/src/ssa_refactor/ssa_gen/mod.rs:21]: We switch in a couple of places -- though let me know if this is on purpose (we just need to have a consistent rule)

**Review by kevaundray — COMMENTED** [review]
Looks good overall -- left some comments and questions

File: crates/noirc_evaluator/src/ssa_refactor/ssa_gen/mod.rs
Comment (jfecher) [reply crates/noirc_evaluator/src/ssa_refactor/ssa_gen/mod.rs:21]: Ok so I've edited it a bit to make it more consistent:
- `SSA` is now used in all comments with some exceptions. It is the most correct since SSA is an acronym for single-static assignment.
- `Ssa` is still used for the `struct Ssa { ... }` type since it follows the casing rules of only upper-casing the first letter in the acronym in case of acronyms. Likewise, any comments referring specifically to this type rather than the SSA concept in general can keep the Ssa spelling.
- `ssa` in lowercase is used only where all lowercase letters are expected like in function names such as `generate_ssa`. Comments in these functions should still refer to converting to "SSA" or "SSA form" in caps.

File: crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs
Comment (jfecher) [reply crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs:7]: That is correct, yes.

This is necessary for non-pure instructions like `Instruction::Load` which may return different results if they're used twice in different spots. Since we attach instruction results to the id of the instruction, their ids must be different to allow different results.

File: crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs
Comment (jfecher) [reply crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs:247]: It is not necessarily useful, but since this would be the behavior if this function was called after a terminator was set, I thought it better to document it than leave it unspecified. The reasoning for this is we store instructions separately from the terminator `{ instructions: Vec<InstructionId>, terminator: Option<TerminatorInstruction> }` so if we're just pushing to the instructions Vec it is naturall the terminator stays at the end rather than somehow winding up in the middle of the Vec - which the type system wouldn't allow in the first place.

File: crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs
Comment (jfecher) [reply crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs:71]: Changed

**Comment by kevaundray** [comment]
Looks good --going to hold this off until #1226 is merged (about 10 minutes)

**Review by kevaundray — APPROVED** [review]

**Pull request merged**
